features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewVersion/NewVersion.jsx

```diff
@@ -259,6 +285,30 @@ class CreateNewVersion extends React.Component {
                                         autoFocus
                                     />
                                 </FormControl>
+                                {api.serviceInfo && (
+                                    <FormControl margin='normal' className={classes.FormControlOdd}>
+                                        <TextField
+                                            id='version-selector'
+                                            select
+                                            label={(
+                                                <FormattedMessage
+                                                    id='Apis.Details.NewVersion.NewVersion.service.version'
+                                                    defaultMessage='Service Version'
+                                                />
+                                            )}
+                                            name='selectVersion'
+                                            onChange={this.handleServiceVersionChange()}
+                                            margin='dense'
+                                            variant='outlined'
+                                        >
+                                            {this.state.versionList && this.state.versionList.map((item) => (
```

Comment:
need to use a key here in the < MenuItem/>

Reply:
Fixed